test/test_scenario_perception_pipeline.cpp

```diff
@@ -7,92 +7,165 @@
+{
+}
 
-bool ScenarioPerceptionPipeline::go_to_poses(std::vector<geometry_msgs::msg::Pose> pose_list) {
-    for (auto& pose : pose_list) {
-        move_group_interface_ptr_->setPoseTarget(pose);
+std::tuple<int, int> ScenarioPerceptionPipeline::runTestCase(const nav_msgs::msg::Path& test_case)
+{
+  int success_number = 0;
+  int failure_number = 0;
+  for (auto& pose_stamped : test_case.poses)
+  {
+    bool is_successful = sendTargetPose(pose_stamped.pose);
+    if (is_successful)
+      success_number++;
+    else
+      failure_number++;
+  }
+
+  return { success_number, failure_number };
+}
 
-        moveit::planning_interface::MoveGroupInterface::Plan plan_msg;
-        auto const ok = static_cast<bool>(move_group_interface_ptr_->plan(plan_msg));
+bool ScenarioPerceptionPipeline::sendTargetPose(const geometry_msgs::msg::Pose& target_pose)
+{
+  move_group_interface_ptr_->setPoseTarget(target_pose);
 
-        if (ok) {
-            move_group_interface_ptr_->execute(plan_msg);
-        } else {
-            return false;
-        }
-    }
+  moveit::planning_interface::MoveGroupInterface::Plan plan_msg;
+  const auto ok = static_cast<bool>(move_group_interface_ptr_->plan(plan_msg));
 
+  if (ok)
+  {
+    move_group_interface_ptr_->execute(plan_msg);
     return true;
+  }
+  else
+  {
+    return false;
+  }
 }
 
-class ScenarioPerceptionPipelineFixture : public benchmark::Fixture {
+class ScenarioPerceptionPipelineTestCaseCreator
+{
+private:
+  static inline std::vector<nav_msgs::msg::Path> test_cases_ = {};
+
 public:
-    std::vector<geometry_msgs::msg::Pose> example_pose_list_;
-    std::shared_ptr<rclcpp::Node> node_;
-    std::shared_ptr<MoveGroupInterface> move_group_interface_ptr_;
-
-    void SetUp(::benchmark::State & state) {
-        geometry_msgs::msg::Pose target_pose1_msg;
-        target_pose1_msg.orientation.w = 1.0;
-        target_pose1_msg.position.x = -0.7;
-        target_pose1_msg.position.y = 0.0;
-        target_pose1_msg.position.z = 0.5;
-
-        geometry_msgs::msg::Pose target_pose2_msg;
-        target_pose2_msg.orientation.w = 1.0;
-        target_pose2_msg.position.x = 0.4;
-        target_pose2_msg.position.y = 0.0;
-        target_pose2_msg.position.z = 0.7;
-
-        for (int i = 0; i < 10; i++) {
-            example_pose_list_.push_back(target_pose1_msg);
-            example_pose_list_.push_back(target_pose2_msg);
-        }
-
-        if (node_.use_count() == 0) {
-            node_ = std::make_shared<rclcpp::Node>(
-        "test_scenario_perception_pipeline_node", rclcpp::NodeOptions().automatically_declare_parameters_from_overrides(true)
-    );
-        }
-
-        if (move_group_interface_ptr_.use_count() == 0) {
-            move_group_interface_ptr_ = std::make_shared<MoveGroupInterface>(node_, "panda_arm");
-        }
+  static void createTestCases()
+  {
+    const std::string yaml_file_path = ament_index_cpp::get_package_share_directory("moveit_middleware_benchmark") +
```

Comment:
A good best practice is to put constants at the beginning of a file (after the includes) into an anonymous namespace